### PR TITLE
Ignore linux for version range checks.

### PR DESCRIPTION
**Pull request by BrianAllred**
Patch fixes issue #2103.

### CODE REVIEW

**Comment by BrianAllred** [comment]
Ha, that was how I originally had it and second guessed myself.

**Comment by jostrander** [comment]
I'd defer to @MarshallOfSound before you change it, he might not mind. Just quickly glancing over it, it seems to be an appendage without explanation.

File: src/renderer/ui/components/generic/PlatformSpecific.js [hunk 1]
```diff
@@ -23,7 +23,7 @@ export default class PlatformSpecific extends Component {
     if (process.platform === this.props.platform) {
       if (!this.props.versionRange) return this.props.children;
 
-      if (semver.validRange(this.props.versionRange) && semver.satisfies(osVersion, this.props.versionRange)) {
+      if ((semver.validRange(this.props.versionRange) && semver.satisfies(osVersion, this.props.versionRange)) || (osVersion === null && process.platform === 'linux')) {
```
Comment (MarshallOfSound) [inline src/renderer/ui/components/generic/PlatformSpecific.js]: If `osVersion` is `null` we should not render regardless of platform.  Also this appears to short-circuit the wrong way.  The `semver.satisfies` check will called on `null` still.

Reverse the "or" order and don't check the platform 👍 

Trying to keep this components in a state where they could be published separately

**Comment by voria** [comment]
This does not fix the problem. As I described in the issue, the problem is mainly at lines 5-6 of the file `src/renderer/ui/components/generic/PlatformSpecific.js`, where `parsedOSVersion` is `null` and then there is an attempt to read its properties, causing an uncaught error.

**Comment by BrianAllred** [comment]
@voria Interesting, I had the white screen issue and this fixed it for me. Maybe when I make @MarshallOfSound's suggested changes, it'll fix it for you.

**Comment by SriRamanujam** [comment]
fwiw this patch does fix #2103 for me on git master, thank you!

**Pull request merged**